test/kubernetes_unittest.cc

```diff
@@ -137,4 +147,74 @@ TEST_F(KubernetesTest, ComputePodAssociations) {
       ComputePodAssociations(reader, pod->As<json::Object>());
   EXPECT_EQ(expected_associations->ToString(), associations->ToString());
 }
+
+TEST_F(KubernetesTest, GetContainerMetadata) {
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: TestClusterName\n"
+    "KubernetesClusterLocation: TestClusterLocation\n"
+    "MetadataApiResourceTypeSeparator: \".\"\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+  const auto m = GetContainerMetadata(
+      reader,
+      json::object({
```

Comment:
I would factor this call out to a `pod` local variable...

Reply:
Updated